libraries/chain/controller.cpp

```diff
@@ -1721,7 +1729,6 @@ struct controller_impl {
             auto branch = fork_db.fetch_branch_from_head();
             int rev = 0;
             for( auto i = branch.rbegin(); i != branch.rend(); ++i ) {
-               if( check_shutdown() ) break;
```

Comment:
It seems that this makes the replay of reversible blocks uninterruptible. Is this what we want?

I wonder if instead, when the replay is interrupted, we shouldn't skip the `fork_db_reset_root_to_chain_head();`?

Reply:
I think it is fine to not allow reversible block replay to be interrupted. Pre-savanna it is often around 300 blocks, post-savanna it should only be around 2 blocks. Either way that should not take very long to replay and it simplifies the removal of the forkdb.

Reply:
> post-savanna it should only be around 2 blocks

unless part of the network is down and 1/3+ of the finalizers are not voting.